server/server.go

```diff
@@ -363,3 +371,20 @@ func (svr *Server) GetStoreStats(bt store.BackendType) (*store.Stats, error) {
 
 	return nil, fmt.Errorf("store not found")
 }
+
+type HandlerError struct {
+	Err  error
+	Meta *commitments.CommitmentMeta
+}
+
+func (e *HandlerError) Error() string {
+	return fmt.Sprintf("HandlerError: %s", e.Err.Error())
+}
+
+// Helper function to create a new HandlerError
+func NewHandlerError(err error, meta *commitments.CommitmentMeta) HandlerError {
+	return HandlerError{
+		Err:  err,
+		Meta: meta,
+	}
+}
```

Comment:
Let's move this to some other file `errors.go`

Reply:
No this is perfect! Just separate file makes it easier but no need to complicate things with separate package until this error is needed in some other package.